### PR TITLE
[1.29] docs: Change reverse proxy to proxy in man page

### PR DESCRIPTION
**Pull request by jvlcek**
This is the backport of https://github.com/candlepin/subscription-manager/pull/3443

Change "reverse proxy" in rhsm.conf man page to simply "proxy"

Fixes: https://issues.redhat.com/browse/RHEL-52660
Fixes: https://issues.redhat.com/browse/CCT-682

### CODE REVIEW

**Comment by jvlcek** [comment]
@pinotree Is there anything else I need to do to move this forward?

**Review by DuckBoss — CHANGES_REQUESTED** [review]
Hi, thanks for your updates! 

It LGTM but I have a small request before approving.
Can you update the commit title/message to follow the [conventional commits](https://www.conventionalcommits.org/) spec that we use?

tl;dr - updating the commit title to something like: 
`docs: update proxy details in man page` and mentioning the related
jira card ID in the message.

**Comment by jvlcek** [comment]
> Hi, thanks for your updates!
> 
> It LGTM but I have a small request before approving. Can you update the commit title/message to follow the [conventional commits](https://www.conventionalcommits.org/) spec that we use?
> 
> tl;dr - updating the following:
> 
> 1. commit title to something like:
>    `docs: update proxy details in man page`
> 2. mentioning the related jira card ID in the message:
>    `Card ID: CCT-....`
> 3. mentioning the card is cherry-picked:
>    `(cherry-picked from commit ...)`

Thank you for the help with this @DuckBoss 

The git log now shows the below. Please let me know if this needs further adjustment of is now good.

```
% git log | head
commit 70077dc6198611696cdee107e4fdf9f7bbecc7bd
Author: Joe VLcek <jvlcek@redhat.com>
Date:   Wed Aug 21 11:51:39 2024 -0400

    docs: update proxy details in man page
    
    Card ID: CCT-682
    
    cherry-pick from fb3d58552
```

**Review by DuckBoss — APPROVED** [review]
LGTM now, thanks!
